libs/vkd3d/resource.c

```diff
@@ -2502,6 +2551,30 @@ HRESULT d3d12_resource_create_placed(struct d3d12_device *device, const D3D12_RE
         object->res.va = object->mem.resource.va;
     }
 
+    if (vkd3d_resource_can_be_vrs(device, &heap->desc.Properties, desc))
```

Comment:
These code paths really should be hoisted out to a helper functions.

Reply:
Done.